javalib/src/main/scala/java/nio/channels/Channels.scala

```diff
@@ -1,9 +1,94 @@
 package java.nio
 package channels
 
+import java.io.{InputStream, OutputStream, Reader, Writer}
+import java.nio.charset.{Charset, CharsetDecoder, CharsetEncoder}
+import java.util.Objects
+import scalanative.annotation.stub
+
 object Channels {
-  @scalanative.annotation.stub
-  def newChannel(
-      is: java.io.InputStream
-  ): java.nio.channels.ReadableByteChannel = ???
+  def newInputStream(channel: ReadableByteChannel): InputStream = {
+    Objects.requireNonNull(channel, "ch")
+    new InputStream {
+      private lazy val buffer = ByteBuffer.allocate(1)
+
+      override def read(buf: Array[Byte], offset: Int, count: Int): Int = {
+        channel.read(ByteBuffer.wrap(buf, offset, count))
+      }
+
+      override def read(): Int = {
+        buffer.position(0)
+        val read = channel.read(buffer)
+        if (read <= 0) read
+        else buffer.get(0) & 0xff
+      }
+
+      override def close(): Unit =
+        channel.close()
+    }
+  }
+
+  def newOutputStream(channel: WritableByteChannel): OutputStream = {
+    Objects.requireNonNull(channel, "ch")
+    new OutputStream {
+      private lazy val buffer = ByteBuffer.allocate(1)
+      override def write(b: Int): Unit = {
+        buffer.position(0)
+        buffer.put(0, b.toByte)
+        channel.write(buffer)
+      }
+      override def write(b: Array[Byte], off: Int, len: Int): Unit = {
+        channel.write(ByteBuffer.wrap(b, off, len))
+      }
+      override def close(): Unit =
+        channel.close()
+    }
+  }
+
+  @stub
+  def newInputStream(ch: AsynchronousByteChannel): InputStream = ???
```

Comment:
Is this, and other `@stub`s, actually required? It would be better not to declare them at all.

Overall, we should get to a state where we have no more `@stub` method left anywhere. Either it's implemented correctly, or it's not there.

Reply:
When methods will be declared as stubs and `linkStubs` in NativeConfig is set to `false` (default) user can get information about missing methods at compile-time, instead of getting unpleasant surprises in the runtime. Based on that it might be better to keep the annotations until we will provide the actual implementation of async channels.

Reply:
Ah, I meant to ask whether declaring those methods is required in the first place. Can't we just remove `newInputStream` altogether? As well as all the other things annotated with `@stub`.

Reply:
Since https://github.com/scala-native/scala-native/pull/2162 was merged we should be able to remove them. They should be implicitly treated as stubs in such case.